libcontainer/container_linux.go

```diff
@@ -285,27 +285,12 @@ func (c *linuxContainer) Start(process *Process) error {
 		}
 
 		if c.config.RootfsUidShiftType == sh.Chown {
-
-			cloneRootfs, err := c.rootfsCloningRequired()
-			if err != nil {
-				return err
-			}
-
-			if cloneRootfs {
-				_, err := c.sysMgr.CloneRootfs(c.config.Rootfs)
-				if err != nil {
-					return newSystemErrorWithCause(err, "failed to clone rootfs")
-				}
-
-				c.config.RootfsCloned = true
-
+			if c.config.RootfsCloned {
```

Comment:
Looks like you're dropping the extra sysbox-runc <-> sysbox-mgr interaction that you created in the previous commits, right? I mean, you're now requesting sysbox-mgr to create the clone rootfs in one of the existing interactions...

Reply:
Yes that's right. It's now in `ConvertSpec()` in `libsysbox/syscont/spec.go`:

```
1190 │    rootfsCloned := false                                                                                                                                                                                                                                                                                               
1191 │    if rootfsUidShiftType == sh.Chown {                                                                                                                                                                                                                                                                                 
1192 │       rootfsCloned, err = cfgRootfsCloning(spec, sysMgr)                                                                                                                                                                                                                                                               
1193 │       if err != nil {                                                                                                                                                                                                                                                                                                  
1194 │          return sh.NoShift, sh.NoShift, false, err                                                                                                                                                                                                                                                                     
1195 │       }                                                                                                                                                                                                                                                                                                                
1196 │    } 
```

The reason I moved it there is because the sysbox-mgr needs to know if the rootfs is cloned as early as possible, so that it can do other setups with this knowledge in hand.